internal/powerplatform/services/licensing/resource_billing_policy.go

```diff
@@ -227,22 +244,31 @@ func (r *BillingPolicyResource) Read(ctx context.Context, req resource.ReadReque
 }
 
 func (r *BillingPolicyResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	var plan *BillingPolicyResourceModel
-
 	tflog.Debug(ctx, fmt.Sprintf("UPDATE RESOURCE START: %s", r.ProviderTypeName))
 
+	var plan *BillingPolicyResourceModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	var state *BillingPolicyResourceModel
 	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
 	if plan.Name.ValueString() != state.Name.ValueString() ||
 		plan.Status.ValueString() != state.Status.ValueString() {
 
+		updateTimeout, diags := plan.Timeouts.Update(ctx, 20*time.Minute)
```

Comment:
All these hard-coded time durations would probably be better managed in a variable somewhere.

Reply:
This is a default and can be overridden by a variable/property on the resource.